File: src/main/java/net/theevilreaper/vulpes/generator/util/FileHelper.java
package net.theevilreaper.vulpes.generator.util;

import org.jetbrains.annotations.ApiStatus;
import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.io.IOException;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;
import java.util.zip.ZipEntry;
import java.util.zip.ZipOutputStream;

@ApiStatus.Internal
public final class FileHelper {

    public static void zipFile(@NotNull Path temp, @NotNull Path zipFile) {
        try (ZipOutputStream zos = new ZipOutputStream(Files.newOutputStream(zipFile))) {
            File tempFile = temp.toFile();
            List<File> filteredFiles = Files.walk(temp)
                    .map(Path::toFile)
                    .filter(file -> !file.getAbsolutePath().equals(tempFile.getAbsolutePath()))
                    .filter(file -> !file.getName().endsWith(".zip"))
                    .toList();

            for (File file : filteredFiles) {
                String zipFileName = file.getAbsolutePath()
                        .replace(tempFile.getAbsolutePath(), "")
                        .replace(File.separator, "");
                ZipEntry entry = new ZipEntry(zipFileName + (file.isDirectory() ? "/" : ""));
                zos.putNextEntry(entry);
                if (file.isFile()) {
                    Files.copy(file.toPath(), zos);
                }
                zos.closeEntry();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private FileHelper() {
        throw new UnsupportedOperationException("This class cannot be instantiated");
    }
}

File: src/main/kotlin/net/theevilreaper/vulpes/generator/util/FileHelper.kt
package net.theevilreaper.vulpes.generator.util

import java.io.File
import java.nio.file.Path
import java.util.zip.ZipEntry
import java.util.zip.ZipOutputStream
import kotlin.io.path.outputStream


/**
 * @author theEvilReaper
 * @version 1.0.0
 * @since
 **/
object FileHelper {

    fun zipFile(temp: Path, zipFile: Path) {
        ZipOutputStream(zipFile.outputStream()).use { zos ->
            val tempFile = temp.toFile()
            val filteredFiles = tempFile.walkTopDown().filter { it.absolutePath != tempFile.absolutePath }
                .filter { !it.name.endsWith(".zip") }
            filteredFiles.forEach { file ->
                val zipFileName = file.absolutePath.removePrefix(tempFile.absolutePath).removePrefix(File.separator)
                val entry = ZipEntry("$zipFileName${(if (file.isDirectory) "/" else EMPTY_STRING)}")
                zos.putNextEntry(entry)
                if (file.isFile) {
                    file.inputStream().copyTo(zos)
                }
            }
        }
    }
}
